promql/parser/lex_test.go

```diff
@@ -674,6 +674,21 @@ var tests = []struct {
 					{RIGHT_BRACKET, 60, `]`},
 				},
 			},
+			{
+				input: `test:name{on!~"bar"}[4m::]`,
```

Comment:
Can we also have a test case with a step? `test:name{on!~"bar"}[4m::15s]` for example